### DISCUSSION TITLE
Install fails to install help

### DISCUSSION THREAD

**Discussion by William-Purvis** [GrandOrgue/grandorgue#1454] [Developers] [answered]
I've been testing GO 3.10.0.0 without install up to now. I had installed the one from Ubuntu, but wanted the latest version.
I've now deleted the Ubuntu version and ran 'sudo make install' but when I request help I just get the help window with
no content. Can I just copy grandorgue.xml to /usr/local/share/Grandorgue/help?

**Comment by ClarkeBrunt** [comment]
As much for my own interest, why not 'sudo apt-get install GOversion.deb' which installs it properly? That's what I did. I also ran the AppImage version a few times while testing, which was also fine, except for a bit of annoyance with the demo organ included in the package. Since the AppImage unpacks itself to a temporary place with a different name every time, it kept re-registering the demo organ with a different path, so multiple instances of the demo organ appeared in dialogs.

**Reply by William-Purvis** [reply]
That's possible, but I wanted to do my own build and install. I have got round it by downloading  one of the releases and extracting the files and copying them into my version, but I'd like to know if this is a bug or have I done something wrong.

**Comment by kerkovits** [comment, marked as answer]
For building the help on your own by `cmake`, you need package `po4a` and a docbook stylesheet (I do not know the ubuntu equivalent for the latter but it is probably `docbook-xsl` or something like this). These packages become unnecessary after building.

**Reply by William-Purvis** [reply]
I must have missed docbook-xsl when checking the pre requisites. It wasn't installed when I ran cmake, butis in the list of prereqs.
I have deleted the build directory, installed docbook-xsl, po4a was already installed and ran cmake, then make, with output to a 
log file. There are quite a few minor error messages, mostly warnings about variables uninitialised, or subscripts out of bounds.
If anyone wants to chase these up and can provide details.

The help/help subdirectory now contains a lot of .html files, which I assume is browsable help info, but I don't see the .htb files.
However, when I run 'make install' it does list them as being installed and they are now present.

Clearly this was my oversight at the start. Many thanks for the advice.

**Reply by kerkovits** [reply]
The overwhelming majority of these compiler warnings are, in general, raised by not the GrandOrgue source, but by the PortAudio libraries included.

**Comment by oleg68** [comment]
@William-Purvis look at [prepare-debian-based.sh](https://github.com/GrandOrgue/grandorgue/blob/master/build-scripts/for-linux/prepare-debian-based.sh) for a full list of prerequisites necessary for building GO on Ubuntu by you own.

Another way of getting the last (not yet released) versions of GO is to look at [the intermediate github builds](https://github.com/GrandOrgue/grandorgue/actions). You may choose a run (please, choose among build runs, don't chose any pre-commit runs) and get the build artifacts for download.
